File: app/src/main/java/com/example/jebeniis/Adapter.java
package com.example.jebeniis;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.HashMap;
import java.util.List;

public class Adapter extends RecyclerView.Adapter<Adapter.ViewHolder> {
	LayoutInflater inflater;
	List<String> items;
	HashMap<String, String> questions;

	public Adapter(Context context, List<String> items, HashMap<String, String> questions) {
		this.inflater = LayoutInflater.from(context);
		this.items = items;
		this.questions = questions;
	}

	@NonNull
	@Override
	public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		View view = inflater.inflate(R.layout.item, parent, false);
		return new ViewHolder(view);
	}

	@Override
	public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
		String item = items.get(position);
		holder.itemName.setText(item);
		holder.itemView.setOnClickListener(v -> {
			Toast.makeText(v.getContext(), questions.get(item), Toast.LENGTH_LONG).show();
		});
	}

	@Override
	public int getItemCount() {
		return items.size();
	}

	public class ViewHolder extends RecyclerView.ViewHolder {
		TextView itemName;

		public ViewHolder(@NonNull View itemView) {
			super(itemView);

			itemName = itemView.findViewById(R.id.itemName);
		}
	}
}

File: app/src/main/java/com/example/jebeniis/Adapter.kt
package com.example.jebeniis

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.RecyclerView


class Adapter(context: Context?, items: List<String>, questions: HashMap<String, String>) :
    RecyclerView.Adapter<Adapter.ViewHolder>() {
    private var inflater: LayoutInflater
    private var items: List<String>
    private var questions: HashMap<String, String>
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = inflater.inflate(R.layout.item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        holder.itemName.text = item
        holder.itemView.setOnClickListener { v: View ->
            val activity = v.context as AppCompatActivity
            val myFragment: Fragment = ItemFragment()
            activity.supportFragmentManager.beginTransaction()
                .replace(R.id.fragmentContainer, myFragment).addToBackStack(null).commit()
            /*Toast.makeText(
                v.context,
                questions[item],
                Toast.LENGTH_LONG
            ).show()*/
        }
    }

    override fun getItemCount(): Int {
        return items.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var itemName: TextView

        init {
            itemName = itemView.findViewById(R.id.itemName)
        }
    }

    init {
        inflater = LayoutInflater.from(context)
        this.items = items
        this.questions = questions
    }
}